src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/publicApi/PublicLicenceService.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.publicApi
+
+import jakarta.transaction.Transactional
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.publicApi.model.licence.LicenceSummary
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.publicApi.model.licence.LicenceStatus as PublicLicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.publicApi.model.licence.LicenceType as PublicLicenceType
+
+@Service
+class PublicLicenceService(
+  private val licenceRepository: LicenceRepository,
+) {
+
+  @Transactional
+  fun getAllLicencesByCrn(crn: String): List<LicenceSummary> {
+    val licences = licenceRepository.findAllByCrnAndStatusCodeIn(crn, LicenceStatus.IN_FLIGHT_LICENCES)
+
+    val modelLicenceSummary = licences.map {
+      it.transformToPublicLicenceSummary(it.typeCode.mapToPublicLicenceType(), it.statusCode.mapToPublicLicenceStatus())
+    }
+
+    return modelLicenceSummary
+  }
+
+  private fun LicenceType.mapToPublicLicenceType() =
```

Comment:
these transformer functions could be internal to your ToModel file - you then wouldn't need to pass the statuses in here?
so it could be: 

```kt
 return licences.map {  it.transformToPublicLicenceSummary() }
```

Reply:
Will implement in next ticket.